### PR TITLE
Improve controller.go coverage & mock K8s add/remove label calls.

### PR DESCRIPTION
**Pull request by randeepsharma**
# Description
Added UTs to cove more code of controller.go
Added admin.conf file to enable runs on github actions

coverage of controller.go improved to 75.7% from 63.1%
overall coverage is now 77.8% from 73.6%

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
https://github.com/dell/csm/issues/350

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Ran make unit-test and all tests are passing.

### CODE REVIEW

**Review by meggm — DISMISSED** [review]

File: service/mock/k8s/admin.conf [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    server: https://10.247.66.177:6443
```
Comment (Sakshi-dell) [inline service/mock/k8s/admin.conf:5]: IP can be changed since this is a mock request

File: service/step_defs_test.go [hunk 2]
```diff
@@ -3147,3 +3147,145 @@ func (f *feature) iCallDeleteSnapshotIsiService() error {
...
+	accessMode.Mode = csi.VolumeCapability_AccessMode_MULTI_NODE_READER_ONLY
+	capability.AccessMode = accessMode
+	capabilities := make([]*csi.VolumeCapability, 0)
+	capabilities = append(capabilities, capability)
+	parameters := make(map[string]string)
+	parameters[AccessZoneParam] = "System"
+	parameters[IsiPathParam] = "/ifs/data/csi-isilon"
+	req.Parameters = parameters
+	req.VolumeCapabilities = capabilities
+	/*req := new(csi.CreateVolumeRequest)
```
Comment (Sakshi-dell) [inline service/step_defs_test.go:3220]: Are these lines commented deliberately?

File: service/step_defs_test.go [hunk 3]
```diff
@@ -1165,7 +1170,6 @@ func (f *feature) iCallNodeGetInfo() error {
 	f.nodeGetInfoResponse, f.err = f.service.NodeGetInfo(context.Background(), req)
 	if f.err != nil {
 		log.Printf("NodeGetInfo call failed: %s\n", f.err.Error())
-		return f.err
```
Comment (bpjain2004) [inline service/step_defs_test.go:1168]: any reason why errors being ignored here ?

**Review by bpjain2004 — COMMENTED** [review]
Thanks @randeepsharma for improving the coverage.

File: service/mock/k8s/admin.conf (hunk 1 above)
Comment (randeepsharma) [reply service/mock/k8s/admin.conf]: changed to 127.0.0.1:36443

File: service/step_defs_test.go (hunk 3 above)
Comment (randeepsharma) [reply service/step_defs_test.go]: brought it back.

File: service/step_defs_test.go (hunk 2 above)
Comment (randeepsharma) [reply service/step_defs_test.go]: uncommented

**Review by shefali-malhotra — DISMISSED** [review]
LGTM

**Review by Sakshi-dell — COMMENTED** [review]
@randeepsharma Shouldn't fakenodecreater.go be added in mocks. Since it is added only for unit tests?

**Comment by randeepsharma** [comment]
> @randeepsharma Shouldn't fakenodecreater.go be added in mocks. Since it is added only for unit tests?

Agree. Moved it under mocks

**Review by Sakshi-dell — APPROVED** [review]
LGTM

**Review by shefali-malhotra — APPROVED** [review]
LGTM

**Pull request merged**